src/hyperlight_host/src/sandbox/outb.rs

```diff
@@ -93,6 +94,58 @@ pub(super) fn outb_log(mgr: &mut SandboxMemoryManager<HostSharedMemory>) -> Resu
     Ok(())
 }
 
+const ABORT_TERMINATOR: u8 = 0xFF;
+
+thread_local! {
+    static GUEST_ABORT_BUFFER: RefCell<Vec<u8>> = const { RefCell::new(Vec::new()) }
+}
+
+fn outb_abort(data: &[u8]) -> Result<()> {
+    GUEST_ABORT_BUFFER.with(|buf| {
+        let mut buffer = buf.borrow_mut();
+
+        for &b in data {
+            if b == ABORT_TERMINATOR {
+                let full = &buffer[..];
+
+                let guest_error_code = *full.first().unwrap_or(&0);
+                let guest_error = ErrorCode::from(guest_error_code as u64);
+
+                match guest_error {
+                    ErrorCode::StackOverflow => {
```

Comment:
What do you think about making stackoverflow it's own abort?

Reply:
It's simpler that the other aborts, but I don't really see a reason to separate it. This sets precedence to custom handling of different error codes, which I think should be okay to do without creating a new OutBAction.